lib/galaxy/tool_util/linters/outputs.py

```diff
@@ -3,146 +3,242 @@
+            tool_node = tool_xml.getroot()
+        if len(tool_xml.findall("./outputs")) == 0:
+            lint_ctx.warn("Tool contains no outputs section, most tools should produce outputs.", node=tool_node)
+
+
+class OutputsOutput(Linter):
+    @classmethod
+    def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
+        tool_xml = getattr(tool_source, "xml_tree", None)
+        if not tool_xml:
+            return
+        output = tool_xml.find("./outputs/output")
+        if output is not None:
+            lint_ctx.warn("Avoid the use of 'output' and replace by 'data' or 'collection'", node=output)
+
+
+class OutputsNameInvalidCheetah(Linter):
+    @classmethod
+    def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
+        tool_xml = getattr(tool_source, "xml_tree", None)
+        if not tool_xml:
+            return
+        for output in tool_xml.findall("./outputs/data[@name]") + tool_xml.findall("./outputs/collection[@name]"):
+            if not is_valid_cheetah_placeholder(output.attrib["name"]):
+                lint_ctx.warn(
+                    f'Tool output name [{output.attrib["name"]}] is not a valid Cheetah placeholder.', node=output
+                )
 
 
-def lint_output(tool_source: "ToolSource", lint_ctx: "LintContext"):
-    """Check output elements, ensure there is at least one and check attributes."""
-    tool_xml = getattr(tool_source, "xml_tree", None)
-    if tool_xml is None:
-        return
-    profile = tool_source.parse_profile()
-
-    outputs = tool_xml.findall("./outputs")
-    # determine node to report for general problems with outputs
-    tool_node = tool_xml.find("./outputs")
-    if tool_node is None:
-        tool_node = tool_xml.getroot()
-    if len(outputs) == 0:
-        lint_ctx.warn("Tool contains no outputs section, most tools should produce outputs.", node=tool_node)
-        return
-    if len(outputs) > 1:
-        lint_ctx.warn("Tool contains multiple output sections, behavior undefined.", node=outputs[1])
-    num_outputs = 0
-    labels = set()
-    names = set()
-    for output in list(outputs[0]):
-        if output.tag is etree.Comment:
-            continue
-        if output.tag not in ["data", "collection"]:
-            lint_ctx.warn(f"Unknown element found in outputs [{output.tag}]", node=output)
-            continue
-        num_outputs += 1
-        if "name" not in output.attrib:
-            lint_ctx.warn("Tool output doesn't define a name - this is likely a problem.", node=output)
-            # TODO make this an error if there is no discover_datasets / from_work_dir (is this then still a problem)
-        elif not is_valid_cheetah_placeholder(output.attrib["name"]):
-            lint_ctx.warn(
-                f'Tool output name [{output.attrib["name"]}] is not a valid Cheetah placeholder.', node=output
-            )
-
-        name = output.attrib.get("name")
-        if name is not None:
+class OutputsNameDuplicated(Linter):
```

Comment:
This needs to be removed or converted to in info. Outputs with duplicate names are possible and can be used to check content assertions for compressed and uncompressed output.

Reply:
This is actually fine. I confused it with `test/output`.